src/libraries/System.Net.Http/src/System/Net/Http/ByteArrayContent.cs

```diff
@@ -58,6 +59,15 @@ public ByteArrayContent(byte[] content, int offset, int count)
         private protected Task SerializeToStreamAsyncCore(Stream stream, CancellationToken cancellationToken) =>
             stream.WriteAsync(_content, _offset, _count, cancellationToken);
 
+        protected override void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+        {
+            Debug.Assert(stream != null);
```

Comment:
this is a public API; this should throw rather than assert.

Reply:
Fixed